modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -304,6 +305,9 @@ public void onException(ConnectionException exception) {
       return;
     }
 
+    refreshTokenIfNecessary(getConfigurationInstance()
+        .flatMap(configurationInstance -> configurationInstance.getConnectionProvider()).orElse(null), exception);
```

Comment:
add logs stating that the token was refreshed. Same thing goes for all the other similar PRs that you sent